common/persistence/sql/SchemaAwareInterface.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ *
+ * @license GPLv2
+ */
+
+namespace oat\generis\persistence\sql;
+
+interface SchemaAwareInterface
+{
+    /**
+     * Allows a class to adapt the schemas as required
+     * @param SchemaCollection $schemaCollection
+     */
+    public function touchSchemas(SchemaCollection $schemaCollection);
```

Comment:
i think the name should be something like `setSchemas`
```suggestion
    public function setSchemas(SchemaCollection $schemaCollection);
```

Reply:
Change name to provide as the "awareinterface" pattern doesn't apply here